devtools/prettier.py

```diff
@@ -239,6 +246,26 @@ def _format_dataclass(self, value: 'Any', _: str, indent_current: int, indent_ne
             self._stream.write(',\n')
         self._stream.write(indent_current * self._c + ')')
 
+    def _format_sqlalchemy_class(self, value: 'Any', _: str, indent_current: int, indent_new: int):
+        import json
+
+        fields = {}
+        for field in [x for x in dir(value) if not x.startswith('_') and x != 'metadata']:
+            data = value.__getattribute__(field)
+            try:
+                json.dumps(data)
+                fields[field] = data
+            except TypeError:
+                fields[field] = None
+
+        before_ = indent_new * self._c
+        self._stream.write(f'{value.__class__.__name__}(\n')
+        for k, v in fields.items():
+            self._stream.write(f'{before_}{k}=')
+            self._format(v, indent_new, False)
+            self._stream.write(',\n')
+        self._stream.write(indent_current * self._c + ')')
```

Comment:
I'd extract this into a common function `self._format_fields(value, fields)` and then for dataclasses use `self._format_fields(value, asdict(value))`. Looks like it could also be used in `_format_tuples` for namedtuples.